cmd/testctxlint/main.go

```diff
@@ -2,11 +2,34 @@
 package main
 
 import (
+	"fmt"
+	"os"
+
 	"golang.org/x/tools/go/analysis/singlechecker"
 
 	"github.com/icedream/testctxlint"
 )
 
+// Build information set by goreleaser
+var (
+	version = "dev"
+	commit  = "none"
+	date    = "unknown"
+)
+
 func main() {
+	// Handle version flag before singlechecker.Main processes it
+	for i, arg := range os.Args {
+		if arg == "-V" || arg == "-V=full" {
+			fmt.Printf("testctxlint version %s, commit %s, built at %s\n", version, commit, date)
+			os.Exit(0)
+		}
+		// Handle -V=full in case it's passed as separate arguments
+		if arg == "-V" && i+1 < len(os.Args) && os.Args[i+1] == "full" {
+			fmt.Printf("testctxlint version %s, commit %s, built at %s\n", version, commit, date)
+			os.Exit(0)
+		}
```

Comment:
This indeed seems redundant and overcomplicates the flag parsing. Just keep it at a simple `-V`.